packages/api/src/routes/medical/__tests__/mapi/e2e/shared.ts

```diff
@@ -0,0 +1,113 @@
+import { MetriportMedicalApi } from "@metriport/api-sdk";
+import { AxiosInstance } from "axios";
+import { FhirClient } from "../../../../../external/fhir/api/api";
+import { apiCognito, api as apiOSS, baseURL, apiInternal } from "../../../../__tests__/shared";
+import { makeFhirApi } from "../../../../../external/fhir/api/api-factory";
+import { testAccount, Account } from "./account";
+import { Util } from "../../../../../shared/util";
+import { getEnvVarOrFail } from "../../../../../shared/config";
+
+export const ACCOUNT_PATH = "/internal/admin/cx-account";
+export const MAPI_ACCESS = "/internal/mapi-access";
+const GENERATE_KEY = "/generateKey";
+
+export type Apis = {
+  apiOSS: AxiosInstance;
+  medicalApi: MetriportMedicalApi;
+  fhirApi: FhirClient;
+};
+
+export type E2ETest = {
+  apis: Apis;
+  account: Account;
+};
+
+export const setupE2ETest = async (isCreatingAccount?: boolean): Promise<E2ETest> => {
+  if (isCreatingAccount) {
+    const account = await apiInternal.post(ACCOUNT_PATH, testAccount);
+    const apiKey = await apiCognito.post(GENERATE_KEY, null, {
+      headers: {
+        Authorization: account.data.idToken,
+      },
+    });
+    apiOSS.defaults.headers["x-api-key"] = apiKey.data.key;
+
+    const medicalApi = new MetriportMedicalApi(apiKey.data.key, { baseAddress: baseURL });
+    const fhirApi = makeFhirApi(account.data.customer.id);
+
+    await apiOSS.post(`${MAPI_ACCESS}?cxId=${account.data.customer.id}`);
+
+    return {
+      apis: {
+        apiOSS,
+        medicalApi,
+        fhirApi,
+      },
+      account: account.data,
+    };
+  }
+
+  // For dev and prod
+  const testId = getEnvVarOrFail("TEST_ACC_ID");
+  const testApiKey = getEnvVarOrFail("TEST_API_KEY");
+
+  const fhirApi = makeFhirApi(testId);
+  apiOSS.defaults.headers["x-api-key"] = testApiKey;
+  const medicalApi = new MetriportMedicalApi(testApiKey, { baseAddress: baseURL });
+
+  return {
+    apis: {
+      apiOSS,
+      medicalApi,
+      fhirApi,
+    },
+    account: {
+      customer: {
+        id: testId,
+        subscriptionStatus: "active",
```

Comment:
Wondering if we shouldn't get the acc info from the server to make sure its on an expected state, one of the items being the sub status.

Reply:
I dont follow, do you mean instead of using a test acc id to actually fetch it?

Reply:
No, I mean that we're returning `active` on the sub status, but there's a non-zero chance that the account is not really active on the server, right? By returning the response from the server we'd be using the state that better represents the server's.